editor/editor_asset_installer.cpp

```diff
@@ -54,23 +54,56 @@ void EditorAssetInstaller::_item_checked_cbk() {
 	updating_source = true;
 	TreeItem *item = source_tree->get_edited();
 	item->propagate_check(0);
+	_fix_conflicted_indeterminate_state(source_tree->get_root(), 0);
 	_update_confirm_button();
 	_rebuild_destination_tree();
 	updating_source = false;
 }
 
-void EditorAssetInstaller::_check_propagated_to_item(Object *p_obj, int p_column) {
-	TreeItem *affected_item = Object::cast_to<TreeItem>(p_obj);
-	if (!affected_item) {
-		return;
-	}
-
-	Dictionary item_meta = affected_item->get_metadata(0);
-	bool is_conflict = item_meta.get("is_conflict", false);
-	if (is_conflict) {
-		affected_item->set_checked(0, false);
-		affected_item->propagate_check(0, false);
+bool EditorAssetInstaller::_fix_conflicted_indeterminate_state(TreeItem *p_item, int p_column) {
+	// If the item is a parent, determine its state based on its non-conflict children.
+	if (p_item->get_child_count() > 0) {
```

Comment:
```suggestion
	if (p_item->get_child_count() == 0) {
		return false;
	}
```
Prefer early returns.